File: leetcode/algorithms/easy/sum-of-left-leaves.java
/*
    leetcode
    category: "Algorithms"
    problem title: "Sum of Left Leaves"
*/

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    
    private boolean isLeafNode(TreeNode node){
        if(node!=null && node.left==null && node.right==null){
            return true;
        } else {
            return false;
        }
    }
    public int sumOfLeftLeaves(TreeNode root) {
        if(root==null){
            return 0;
        } else {
            return (root.left!=null && isLeafNode(root.left) ? root.left.val : 0)+sumOfLeftLeaves(root.left)+sumOfLeftLeaves(root.right); 
        }
    }
}

File: leetcode/algorithms/tree/sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isLeafNode(self,root):
        return root!=None and root.left==None and root.right==None
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        
        if root==None:
            return 0;
        
        
        return  (root.left.val if self.isLeafNode(root.left) else self.sumOfLeftLeaves(root.left)) + self.sumOfLeftLeaves(root.right);